Sources/NIO/NonBlockingFileIO.swift

```diff
@@ -225,4 +225,41 @@ public struct NonBlockingFileIO {
         }
         return p.futureResult
     }
+
+    /// Open the file at `path` on a private thread pool which is separate from any `EventLoop` thread.
+    ///
+    /// This function will return (a future) of the `FileHandle` associated with the file opened and a `FileRegion`
+    /// comprising of the whole file. The caller must close the returned `FileHandle` when its no longer needed.
+    ///
+    /// - note: The reason this returns the `FileHandle` and the `FileRegion` is that both the opening of a file as well as the querying of its size are blocking.
+    ///
+    /// - parameters:
+    ///     - path: The path of the file to be opened.
+    ///     - eventLoop: The `EventLoop` on which the returned `EventLoopFuture` will fire.
+    /// - returns: An `EventLoopFuture` containing the `FileHandle` and the `FileRegion` comprising the whole file.
+    public func openFile(path: String, eventLoop: EventLoop) -> EventLoopFuture<(FileHandle, FileRegion)> {
+        let p: EventLoopPromise<(FileHandle, FileRegion)> = eventLoop.newPromise()
+        self.threadPool.submit { shouldRun in
+            guard case shouldRun = BlockingIOThreadPool.WorkItemState.active else {
+                p.fail(error: ChannelError.ioOnClosedChannel)
+                return
+            }
```

Comment:
Should we have a convenience wrapper around `submit` that adds this boilerplate?

Reply:
proposed an API for that now.